src/model_io/urdf/tests/ModelExporterUnitTest.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2017 Fondazione Istituto Italiano di Tecnologia
+ * Authors: Silvio Traversaro
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ *
+ */
+
+#include "testModels.h"
+
+#include <iDynTree/Core/TestUtils.h>
+
+
+#include <iDynTree/Model/Model.h>
+#include <iDynTree/ModelIO/ModelLoader.h>
+#include <iDynTree/ModelIO/ModelExporter.h>
+
+
+#include <cassert>
+#include <cstdio>
+#include <cstdlib>
+#include <algorithm>
+#include <fstream>
+
+using namespace iDynTree;
+
+
+void checkImportExportURDF(std::string fileName)
+{
+    // Import, export and re-import a URDF file, and check that the key properties of the model are mantained
+    ModelLoader mdlLoader;
+    bool ok = mdlLoader.loadModelFromFile(fileName, "urdf");
+    Model model = mdlLoader.model();
+    ASSERT_IS_TRUE(ok);
+
+    std::cerr << "Model loaded from " << fileName << std::endl;
+    std::ifstream original_urdf(fileName);
+
+    if (original_urdf.is_open()) {
+        std::cout << original_urdf.rdbuf();
+    }
+
+    std::string urdfString;
+    ModelExporter mdlExporter;
+    ok = mdlExporter.init(model, SensorsList());
+    ASSERT_IS_TRUE(ok);
+    ok = mdlExporter.exportModelToString(urdfString);
+    ASSERT_IS_TRUE(ok);
+
+    std::cerr << "Model serialized back to xml " << std::endl << urdfString << std::endl;
+
+    ModelLoader mdlLoaderReloaded;
+
+    ok = mdlLoaderReloaded.loadModelFromString(urdfString);
+    ASSERT_IS_TRUE(ok);
+    Model modelReloaded = mdlLoaderReloaded.model();
+
+    ASSERT_EQUAL_DOUBLE(model.getNrOfLinks(), modelReloaded.getNrOfLinks());
+    ASSERT_EQUAL_DOUBLE(model.getNrOfJoints(), modelReloaded.getNrOfJoints());
+    ASSERT_EQUAL_DOUBLE(model.getNrOfDOFs(), modelReloaded.getNrOfDOFs());
+
```

Comment:
Is there any special reason why don't you check also for the sensors?

Reply:
Because at the moment they are not exported, so the test would fail. Relevant test will be added as soon as  sensors exporting is added.